managed/services/inventory/agents.go

```diff
@@ -158,6 +159,54 @@ func (as *AgentsService) Get(ctx context.Context, id string) (inventorypb.Agent,
 	return res, e
 }
 
+// Logs by Agent ID.
+//nolint:unparam
+func (as *AgentsService) Logs(ctx context.Context, id string) ([]string, error) {
+	agent, err := models.FindAgentByID(as.db.Querier, id)
+	if err != nil {
+		return nil, err
+	}
+
+	var pmmAgentID *string
+
+	switch agent.AgentType {
+	case models.PMMAgentType:
+		pmmAgentID = &agent.AgentID
+	case models.NodeExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.MySQLdExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.MongoDBExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.PostgresExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.ProxySQLExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.RDSExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.AzureDatabaseExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.QANMySQLPerfSchemaAgentType:
+		pmmAgentID = agent.PMMAgentID
+	case models.QANMySQLSlowlogAgentType:
+		pmmAgentID = agent.PMMAgentID
+	case models.QANMongoDBProfilerAgentType:
+		pmmAgentID = agent.PMMAgentID
+	case models.QANPostgreSQLPgStatementsAgentType:
+		pmmAgentID = agent.PMMAgentID
+	case models.QANPostgreSQLPgStatMonitorAgentType:
+		pmmAgentID = agent.PMMAgentID
+	case models.ExternalExporterType:
+		pmmAgentID = agent.PMMAgentID
+	case models.VMAgentType:
+		pmmAgentID = agent.PMMAgentID
+	default:
+		panic(fmt.Errorf("unhandled inventory Agent type %s", agent.AgentType))
```

Comment:
can we return error instead of panicing?

Reply:
yes, i will return

Reply:
@ritbl return `error` done